packages/contract/src/pay/DaimoPayRelayer.sol

```diff
@@ -2,83 +2,122 @@
 pragma solidity ^0.8.12;
 
 import "openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
-import "openzeppelin-contracts/contracts/access/Ownable2Step.sol";
 import "openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
 
 import "./DaimoPay.sol";
 import "./TokenUtils.sol";
 
 /*
  * Relayer contract that funds completes DaimoPay intents.
  */
-contract DaimoPayRelayer is Ownable2Step {
+contract DaimoPayRelayer is AccessControl {
     using SafeERC20 for IERC20;
 
-    constructor(address _owner) Ownable(_owner) {}
+    bytes32 public constant RELAYER_ROLE = keccak256("RELAYER_ROLE");
+
+    event SwapAndTip(
+        address indexed requiredTokenIn,
+        uint256 suppliedAmountIn,
+        address indexed requiredTokenOut,
+        uint256 swapAmountOut,
+        uint256 maxPreTip,
+        uint256 maxPostTip
```

Comment:
might be worth calling this `maxIntentTip` and `maxStartTip` for clarity

Reply:
`maxStartTip` would be the post-tip and `maxIntentTip` would be the pre-tip?

personally, i prefer `maxPreTip` and `maxPostTip`. i think this naming makes more sense at the function-level context, whereas `maxStartTip` and `maxIntentTip` requires global-level context to make sense